### ISSUE TITLE
how to get this working on a server? 

### ISSUE THREAD

**Issue by henryw374** [dtreskunov/wta-wdc#1]
Hi, 

I am trying to get a wdc working on a tableau server and so far Tableau support team have been unable to help - even after many calls and emails - so I'm reaching out here in desperation because this connector is given as an example in the tableau docs https://help.tableau.com/current/server/en-us/datasource_wdc.htm 

What I'm after is really any working example command that I can get a wdc refreshing on a server. 

The docs give this as the example command:

`tsm data-access web-data-connectors add --name "WTA WDC" --url https://dtreskunov.github.io:80/wta-wdc/ --secondary https://dtreskunov.github.io/(.*)`

The problem I see with this command is that the port is not going to be 80, but 443. Also the wdc loads bootstrap from a cdn - so wouldn't that need to be on the whitelist too? 

Also this wdc doesn't seem to work in my 2020 version of tableau desktop - I don't know what's wrong.

If this example wdc could be updated and the docs changed to reflect what would actually be needed to get it working on a server that would be a great starting point for us.

**Comment by henryw374** [comment]
I have got wdc refreshing on a tableau server now - but still if the docs relating to this wdc could be corrected I think that could save a lot of headaches.

**Comment by maxrandolph** [comment]
Hi @henryw374, could you share what config you're using that led to a resolution of the issue?

**Comment by maxrandolph** [comment]
Also, are you running your Tableau Server instance on Linux or Windows?

**Comment by henryw374** [comment]
hi. this is on windows. 

I didn't get this particular wdc working but my own. 

I used a json entity, as describe here: https://help.tableau.com/current/server/en-us/entity_wdc.htm but the url format needed to be `"(https://dtreskunov.github.io/(.*))"` - so a bit different to what is shown there. 

For this wdc to work, the secondary whitelist would need to include all of the resources loaded in the html, like bootstrap, tableau js file & etc

**Comment by maxrandolph** [comment]
Yeah, I agree that the docs could use some additional clarity to specify that all the resources including those loaded through the `<script>` tags need to be added to the secondary whitelist. I think they could also use additional clarity with the differences in the globbing wildcard differences between running on windows vs. linux. It seems like only windows supports the regex style i.e. `(.*)`, while linux is just an asterisk `https://dtreskunov.github.io/*`

It does seem like I'm able to repro the issue that the server extract refreshes time out without errors when they haven't been included in that secondary whitelist. I do see some server log entries indicating the failed requests for those resources in the `<script>` tags, but having to wait two hours (by default server config) for the extract refresh to fail without a clear error indication is not great UX. In my experience, the requests only lead to an immediate failure + error when they're not whitelisted and the flow is in either the `getSchema` or the `getData` phases of the WDC.

While I don't have an exact timeline, we can hopefully improve this soon.
